packages/app/src/components/Normal/FloatingActionButton/index.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { ReactNode, useCallback, useState } from "react"
+import { FC } from "react"
+import { enhance } from "src/utilities/essentials"
+import { FloatingActionButtonFallback } from "./fallback"
+import { x } from "@xstyled/styled-components"
+import { Minus } from "@styled-icons/boxicons-regular/Minus"
+import { Add } from "@styled-icons/material-sharp/Add"
+import { V } from "src/styles/styleFragments"
+import { PlaylistPlay } from "styled-icons/material-sharp"
+import { Settings } from "styled-icons/material-sharp"
+import { Information } from "styled-icons/ionicons-sharp"
+
+export type Menu = {
+  icon: ReactNode
+  menuType: MenuType
+}
+
+enum MenuType {
+  DEFAULT,
+  PLAYLIST,
+  SETTINGS,
+  INFO,
+}
+
+export type FloatingActionButtonImpureProps = {
+  //
+}
+
+export const FloatingActionButtonImpure: FC<FloatingActionButtonImpureProps> =
+  enhance<FloatingActionButtonImpureProps>(() => {
+    const [isActive, setIsActive] = useState(false)
+
+    const onClickFloatingButton = useCallback(() => {
+      setIsActive((prev) => !prev)
+    }, [])
+
+    const menus = [
+      {
+        icon: <PlaylistPlay />,
+        menuType: MenuType.PLAYLIST,
+      },
+      {
+        icon: <Settings />,
+        menuType: MenuType.SETTINGS,
+      },
+      {
+        icon: <Information />,
+        menuType: MenuType.INFO,
+      },
+    ]
```

Comment:
```suggestion
    const menus = useMemo(() => [
      {
        icon: <PlaylistPlay />,
        menuType: MenuType.PLAYLIST,
        onClick: () => {
        }
      },
      {
        icon: <Settings />,
        menuType: MenuType.SETTINGS,
                onClick: () => {
        }
      },
      {
        icon: <Information />,
        menuType: MenuType.INFO,
               onClick: () => {
        }
      },
    ], [])
```

밑에 onClickMenu 있을 필요 없이 onClick까지 여기에 넣어서 iterate하면 간단해질듯.
불필요한 렌더를 방지하기 위한 useMemo 추가

Reply:
그리고 이렇게 되면 menuType도 필요없지 않나 하는 생각이 드네. 나중에 필요할지도 모르겠지만

Reply:
menuType 삭제하고 onclick 각각 구현해주도록 했습니다. 
 useMemo사용.